code/datums/components/chem_booster.dm

```diff
@@ -384,9 +389,15 @@
 /datum/component/chem_booster/proc/update_resource(amount)
 	var/amount_added = min(resource_storage_max - resource_storage_current, amount)
 	resource_storage_current = max(resource_storage_current + amount_added, 0)
-	wearer.overlays -= resource_overlay
-	resource_overlay.alpha = resource_storage_current/resource_storage_max*255
-	wearer.overlays += resource_overlay
+	//Check if reagent level was empty and now is not, or if it was not empty but now is
+	if(CONFIG_GET(flag/armor_hud) && ishuman(wearer) && ((!resource_storage_current && amount_added) || resource_storage_current == amount_added))
+		hud_resource_update()
```

Comment:
dont you already have a signal for this?

Reply:
The signal is only sent when equipping armor with the Vali module. Needed for the first time icon display.